modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/AppendProcessorTests.java

```diff
@@ -123,21 +124,38 @@ public void testConvertScalarToList() throws Exception {
     }
 
     public void testAppendMetadata() throws Exception {
-        //here any metadata field value becomes a list, which won't make sense in most of the cases,
+        // here any metadata field value becomes a list, which won't make sense in most of the cases,
         // but support for append is streamlined like for set so we test it
         IngestDocument.MetaData randomMetaData = randomFrom(IngestDocument.MetaData.values());
-        List<String> values = new ArrayList<>();
+        List<Object> values = new ArrayList<>();
         Processor appendProcessor;
-        if (randomBoolean()) {
-            String value = randomAlphaOfLengthBetween(1, 10);
-            values.add(value);
-            appendProcessor = createAppendProcessor(randomMetaData.getFieldName(), value);
+        int valuesSize = randomIntBetween(0, 10);
+        if (randomMetaData == IngestDocument.MetaData.VERSION) {
```

Comment:
Can version and version_type be tested in a dedicated test? I think that makes these tests easier to read.